### PR TITLE
Show total_bytes and total_rows in system tables for RocksDB storage

### PR DESCRIPTION
**Pull request by AVMusorin**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Show `total_bytes` and `total_rows` in system tables for RocksDB storage

Follows: https://github.com/ClickHouse/ClickHouse/pull/55806
cc @canhld94 @vdimir @azat

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit c6ac6439efacc4c9c9719e77287a53cebe670a05 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56816/c6ac6439efacc4c9c9719e77287a53cebe670a05/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6883324623">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56816/c6ac6439efacc4c9c9719e77287a53cebe670a05/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Storages/RocksDB/EmbededRocksDBSettings.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    class AbstractConfiguration;
+}
+
+
+namespace DB
+{
+class ASTStorage;
+class ASTSetQuery;
+
+#define LIST_OF_ROCKSDB_SETTINGS(M, ALIAS) \
```
Comment (canhld94) [inline src/Storages/RocksDB/EmbededRocksDBSettings.h:20]: Why do we need to introduce settings for rocksdb?

File: src/Storages/RocksDB/StorageEmbeddedRocksDB.cpp [hunk 2]
```diff
@@ -703,17 +724,26 @@ void registerStorageEmbeddedRocksDB(StorageFactory & factory)
...
+    std::shared_lock lock(rocksdb_ptr_mx);
+    if (!rocksdb_ptr)
+        return {};
+    UInt64 estimated_rows;
+    if (!rocksdb_ptr->GetIntProperty("rocksdb.estimate-num-keys", &estimated_rows))
+        return {};
+    return estimated_rows;
+}
+
+std::optional<UInt64> StorageEmbeddedRocksDB::totalBytes([[maybe_unused]] const Settings & settings) const
```
Comment (canhld94) [inline src/Storages/RocksDB/StorageEmbeddedRocksDB.cpp:738]: Better to just comment `settings`

File: src/Storages/RocksDB/EmbededRocksDBSettings.h (hunk 1 above)
Comment (vdimir) [reply src/Storages/RocksDB/EmbededRocksDBSettings.h:20]: As far as I understand, enabling the `optimize_trivial_approximate_count_query` setting is for determining whether count approximation can be used while querying `system.tables`.
However, I also doubt introducing settings specific to the RocksDB engine. Why can't we just enable the `optimize_trivial_approximate_count_query` setting globally for the server? I assume users might want it enabled or disabled for all tables using the engine.

Comment (AVMusorin) [reply src/Storages/RocksDB/EmbededRocksDBSettings.h:20]: My initial idea was that by default, this setting should be turned on (at least I do not know why it should be avoided not to use this optimization) also, `total_rows` and `total_bytes` should be in the system tables independently of the `optimize_trivial_approximate_count_query` setting.

> Why do we need to introduce settings for rocksdb?

I like how settings related only to one engine are placed in their own settings file.

> However, I also doubt introducing settings specific to the RocksDB engine. Why can't we just enable the optimize_trivial_approximate_count_query setting globally for the server? I assume users might want it enabled or disabled for all tables using the engine.

As I mentioned in the initial idea, I am reckoning that this setting should be turned on by default, but for some reason, a user can disable it. I thought it was easier to do it with a table setting, but if you are counting that a global setting works better, no objections. 

But I agree, now it's unnecessary complexity, I'll rewrite it.

Comment (AVMusorin) [reply src/Storages/RocksDB/EmbededRocksDBSettings.h]: So, I rethought the changes, tried to simplify them, and avoid incompatibilities. 

 Also added a short description in the documentation about `optimize_trivial_approximate_count_query`.

In short, everything works as before except now there is a total_bytes in `system.tables`, total_rows can be shown with optimize_trivial_approximate_count_query setting.

**Review by vdimir — APPROVED** [review]

**Pull request merged**